src/conn/conn_api.c

```diff
@@ -3065,6 +3086,10 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler, const char *c
         __wt_free(session, conn->partial_backup_remove_ids);
 
     if (ret != 0) {
+        if (conn->default_session->event_handler->handle_general != NULL)
+            conn->default_session->event_handler->handle_general(
+              conn->default_session->event_handler, &conn->iface, NULL, WT_EVENT_CONN_CLOSE);
```

Comment:
> If we do want to call the CLOSE handler here, I suggest only doing so if wiredtiger_open() has gotten far enough that we already called the READY handler. I.e., if WT_CONN_MINIMAL or WT_CONN_READY is set. I think it would be confusing for an application otherwise. Also, it would make it unclear what is/isn't available from the handler as there are lots of ways to wind up here early in setup.

Once the application has been signaled that a connection is ready in "minimal" mode, it must be notified when the connection is being closed. Otherwise the application will have session pointers into live WT memory that is going to be freed. But I do agree that only sending a "conn close" event when a "conn minimal" or "ready" event was sent is preferable.